src/errors.ts

```diff
@@ -24,6 +24,17 @@ export abstract class WidgetError extends Error {
   }
 }
 
+abstract class DismissableWidgetError extends WidgetError {
+  constructor(config: WidgetErrorConfig) {
+    super({
+      action: config.action ?? DEFAULT_DISMISSABLE_ERROR_ACTION,
+      header: config.header ?? DEFAULT_DISMISSABLE_ERROR_HEADER,
+      ...config,
```

Comment:
This could be buggy, in the case that action or header are explicitly undefined. They will then be overwritten on L32 to be undefined. Consider

    const a = {
      a: 1,
      ...{a: undefined}
    }